SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -127,3 +137,29 @@ void UReferenceCountedLockingPolicy::OnLockedActorDeleted(AActor* DestroyedActor
 	}
 	ActorToLockingState.Remove(DestroyedActor);
 }
+
+bool UReferenceCountedLockingPolicy::AcquireLockFromDelegate(AActor* ActorToLock, const FString& DelegateLockIdentifier)
+{
+	ActorLockToken LockToken = AcquireLock(ActorToLock, DelegateLockIdentifier);
+	if (LockToken == SpatialConstants::INVALID_ACTOR_LOCK_TOKEN)
+	{
+		UE_LOG(LogReferenceCountedLockingPolicy, Error, TEXT("AcquireLock called from engine delegate returned an invalid token"));
+		return false;
+	}
+
+	check(!DelegateLockingIdentifierToActorLockToken.Contains(DelegateLockIdentifier));
+	DelegateLockingIdentifierToActorLockToken.Add(DelegateLockIdentifier, LockToken);
+	return true;
+}
+
+bool UReferenceCountedLockingPolicy::ReleaseLockFromDelegate(AActor* ActorToRelease, const FString& DelegateLockIdentifier)
+{
+	if (!DelegateLockingIdentifierToActorLockToken.Contains(DelegateLockIdentifier))
+	{
+		UE_LOG(LogReferenceCountedLockingPolicy, Error, TEXT("Executed ReleaseLockDelegate for unidentified delegate lock identifier. Token: %d."), *DelegateLockIdentifier);
```

Comment:
DelegateLockIdentifier is a string - but log expects an int

Reply:
excellent point, well made